pkg/distributor/http.go

```diff
@@ -39,10 +40,15 @@ func (d *Distributor) PushHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	resp, ok := httpgrpc.HTTPResponseFromError(err)
-	if ok {
+	if parseErr, ok := err.(logql.ParseError); ok {
```

Comment:
I'm not sure if i'm just missing this, but I don't see where the distributor code ever throws a `logql.ParseError`? this sounds more like something we would see on the query path, not the write path.

Reply:
I can't tell where the error comes from, but it ends up here. And this works...

Reply:
Presumably, on the write path the "labels" value must be parsed somewhere

Reply:
@slim-bean the `logql.ParseError` comes from here:
https://github.com/grafana/loki/blob/29f08ccb1cde0e26d57d2c95e293eceb4df17976/pkg/distributor/distributor.go#L148

This is actual ingester logic, so it seems good to me?